interactive-etl/README.md

```diff
@@ -0,0 +1,330 @@
+However, you can run the Flink SQL Runner container locally using the command below (make sure to add the `--net=host` flag so the container can see the forwarded job-manager port):
+
+```shell
+podman run -it --rm --net=host quay.io/streamshub/flink-sql-runner:latest /opt/flink/bin/sql-client.sh embedded
+```
+
+If you use docker, you should be able to replace `podman` with `docker` in the command above.
+
+## Tutorial
+
+This tutorial will walk through some basic data exploration and ETL (extract transform load) queries, using Flink SQL.
+The [Flink SQL documentation](https://nightlies.apache.org/flink/flink-docs-release-1.19/docs/dev/table/sql/queries/overview/) contains detailed breakdowns of the various SQL commands and query functions available.
+
+### Source Data Table
+
+The data generator application creates a topic (`flink.sales.records`) containing sales records. 
+The schema for this topic can be seen in the `data-generator/src/main/resources/sales.avsc`:
+
+```avroschema
+{
+   "namespace": "com.github.streamshub.kafka.data.generator.schema",
+   "type": "record",
+   "name": "Sales",
+   "fields": [
+      {"name": "user_id", "type": "string"},
+      {"name": "product_id",  "type": "string"},
+      {"name": "invoice_id",  "type": "string"},
+      {"name": "quantity",  "type": "string"},
+      {"name": "unit_cost",  "type": "string"}
+   ]
+}
+```
+
+In order to interact with the contents of this topic we need to create a table within Flink SQL:
+
+```sql
+CREATE TABLE SalesRecordTable ( 
+    invoice_id STRING, 
+    user_id STRING, 
+    product_id STRING, 
+    quantity STRING, 
+    unit_cost STRING, 
+    `purchase_time` TIMESTAMP(3) METADATA FROM 'timestamp', 
+    WATERMARK FOR purchase_time AS purchase_time - INTERVAL '1' SECOND 
+) WITH ( 
+    'connector' = 'kafka',
+    'topic' = 'flink.sales.records', 
+    'properties.bootstrap.servers' = 'my-cluster-kafka-bootstrap.flink.svc:9092', 
+    'properties.group.id' = 'sales-record-group', 
+    'value.format' = 'avro-confluent', 
+    'value.avro-confluent.url' = 'http://apicurio-registry-service.flink.svc:8080/apis/ccompat/v6', 
+    'scan.startup.mode' = 'latest-offset'
+); 
+```
+
+The [`CREATE`](https://nightlies.apache.org/flink/flink-docs-release-1.19/docs/dev/table/sql/create/) statement will set up a table with the defined fields within the Flink SQL client. 
+Keep in mind that nothing has actually been sent to the Flink cluster at this point, we have just setup where data will go once we run a query.
+The [`WITH`](https://nightlies.apache.org/flink/flink-docs-release-1.19/docs/dev/table/sql/queries/with/) clause allows us to pass required configuration to Flink to allow it to connect to external sources.
+In the query above we are telling Flink to use the Kafka connector to pull data from the specified Kafka cluster and topic and also to talk to Apicurio to find the Apache Avro schema for the values in the messages of that topic.
```

Comment:
This is common across all our demos, right? I think this description is valuable so I wonder how we can share it sensibly? 

- Asciidoc and publish it? 
- Jekyell site?
- manual duplication?
- builder script to pull the markdown together?